From 57d0b7e6823f5db851578328f49da7a8aa7bc287 Mon Sep 17 00:00:00 2001
From: chonepieceyb <1759315491@qq.com>
Date: Thu, 14 Dec 2023 10:39:37 +0800
Subject: [PATCH 3/3] fix libbpf modstops bug

---
 tools/lib/bpf/libbpf.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tools/lib/bpf/libbpf.c b/tools/lib/bpf/libbpf.c
index eac389f7360f..78824afccd01 100644
--- a/tools/lib/bpf/libbpf.c
+++ b/tools/lib/bpf/libbpf.c
@@ -460,7 +460,7 @@ struct bpf_struct_ops {
 	__u32 type_id;
 
 	/*supported for bpf module struct ops*/
-	const char *module_name;
+	char module_name[64];
 };
 
 #define DATA_SEC ".data"
@@ -11505,7 +11505,8 @@ int bpf_map__set_struct_ops_module(struct bpf_map *map, const char *module_name)
 	}
 
 	st_ops = map->st_ops;
-	st_ops->module_name = module_name;
+	strncpy(st_ops->module_name, module_name, sizeof(st_ops->module_name));
+	st_ops->module_name[sizeof(st_ops->module_name) - 1] = '\0';
 	map->map_extra = (__u64)mod_btf->fd << 32;
 	return 0;
 }
-- 
2.34.1

